### PR TITLE
Improve performances of array_key_last

### LINKED ISSUE #293: [improvement] array_key_last
I've note some minor performance issues on array_key_last.
A good work has been done on this pull request #150 

I think we could do a little better with array_slice instead of using both end and key function.

Here I come with [benchmarks](http://sandbox.onlinephpfunctions.com/code/5984e7ea158cbb306a9d36066967ca09e5b77ce3)

```php
function array_key_last(array $array) {
    return (!empty($array)) ? key(array_slice($array, -1, 1, true)) : null;
}
```

### PR DESCRIPTION
**Pull request by jderusse**
Suggested by @donaldinou   (fix #293 )

up to 6 times faster on big arrays like `array_fill(0, 1024*1024, 'hello');`

https://blackfire.io/profiles/05de3d1f-0193-462d-839f-361c4c206a0d/graph

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thank you @jderusse.

File: src/Php73/bootstrap.php [hunk 1]
```diff
@@ -27,5 +27,5 @@ function hrtime($asNum = false) { return p\Php73::hrtime($asNum); }
     function array_key_first(array $array) { foreach ($array as $key => $value) { return $key; } }
 }
 if (!function_exists('array_key_last')) {
-    function array_key_last(array $array) { end($array); return key($array); }
+    function array_key_last(array $array) { key(array_slice($array, -1, 1, true)); }
```
Comment (nicolas-grekas) [inline src/Php73/bootstrap.php:30]: funny that we missed the return statement :)

Comment (nicolas-grekas) [reply src/Php73/bootstrap.php:30]: fixed in df3c863

Comment (stof) [reply src/Php73/bootstrap.php:30]: isn't the testsuite catching this kind of mistakes ?

Comment (nicolas-grekas) [reply src/Php73/bootstrap.php:30]: yes it is, that's how I spotted it

Comment (stof) [reply src/Php73/bootstrap.php:30]: then we have an issue in our process if we merge PRs when the CI tells us it's broken...

Comment (jderusse) [reply src/Php73/bootstrap.php:30]: Sorry about that!

CI were broken at that time https://travis-ci.org/github/symfony/polyfill/jobs/737662800 

Did 2 mistake:
I didn't ran test locally (I often count on CI for that)
I didn't marked the PR as draft

truly sorry

**Pull request merged**